### PR TITLE
Enforce NoSignature witness type size

### LINKED ISSUE #899: Limit the number of characters in NoSignature InputWitness type
`InputWitness::NoSignature` can have arbitrary data. The size should be limited in consensus to avoid spam.

### PR DESCRIPTION
**Pull request by TheQuantumPhysicist**
Closes #899

### CODE REVIEW

**Review by pavel-kokolemin — APPROVED** [review]

File: chainstate/src/detail/chainstateref/mod.rs [hunk 1]
```diff
@@ -634,6 +634,29 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
...
+                    common::chain::signature::inputsig::InputWitness::NoSignature(data) => {
+                        if let Some(inner_data) = data {
+                            if inner_data.len() > self.chain_config.max_no_signature_data_size() {
+                                return Err(
+                                    CheckBlockTransactionsError::NoSignatureDataSizeTooLarge(
+                                        inner_data.len(),
+                                        self.chain_config.max_no_signature_data_size(),
+                                    ),
+                                );
+                            }
```
Comment (iljakuklic) [inline chainstate/src/detail/chainstateref/mod.rs:650]: Tiny nitpick: This could be `ensure!`... saves one level of indentation

**Review by iljakuklic — APPROVED** [review]
+1. Maybe add a simple test? Should not be too hard.

File: chainstate/src/detail/chainstateref/mod.rs (hunk 1 above)
Comment (OBorce) [reply chainstate/src/detail/chainstateref/mod.rs:650]: another option is txs.iter().flat_map(signatures).find_map(match...)

Comment (TheQuantumPhysicist) [reply chainstate/src/detail/chainstateref/mod.rs]: Went with ensure. Thanks.

**Comment by TheQuantumPhysicist** [comment]
> +1. Maybe add a simple test? Should not be too hard.

Done.

**Review by azarovh — APPROVED** [review]

**Pull request merged**